common/transport/amqp/devdoc/sender_link_requirements.md

```diff
@@ -4,73 +4,65 @@
 ### attach(callback: (err?: Error) => void): void
 
-**SRS_NODE_AMQP_SENDER_LINK_16_004: [** The `attach` method shall use the stored instance of the `amqp10.AmqpClient` object to attach a new link object with the `linkAddress` and `linkOptions` provided when creating the `SenderLink` instance. **]**
+
+**SRS_NODE_AMQP_SENDER_LINK_16_004: [** The `attach` method shall use the stored instance of the `rhea` session object to attach a new link object with the combined `linkAddress` and `linkOptions` provided when creating the `SenderLink` instance. **]**
 
 **SRS_NODE_AMQP_SENDER_LINK_16_022: [** The `attach` method shall call the `callback` if the link was successfully attached. **]**
 
-**SRS_NODE_AMQP_SENDER_LINK_16_006: [** The `SenderLink` object should subscribe to the `detached` event of the newly created `amqp10` link object. **]**
+**SRS_NODE_AMQP_SENDER_LINK_06_007: [** The `attach` method shall immediately invoke the `callback` if already in an attached state. **]**
 
-**SRS_NODE_AMQP_SENDER_LINK_16_007: [** The `SenderLink` object should subscribe to the `errorReceived` event of the newly created `amqp10` link object. **]**
+**SRS_NODE_AMQP_SENDER_LINK_06_001: [** If the `detach` method is invoked on the `SenderLink` while still attaching, the SenderLink shall detach.  It will indicate the error to the callback for the `detach` as well as the callback to the `attach` **]**
 
-**SRS_NODE_AMQP_SENDER_LINK_16_008: [** If the `amqp10.AmqpClient` fails to create the link the `callback` function shall be called with this error object. **]**
+**SRS_NODE_AMQP_SENDER_LINK_06_003: [** If the `forceDetach` method is invoked on the `SenderLink` while still attaching, the SenderLink shall detach.  It will indicate the error to the callback to the `attach` and emit an `error` event on the `SenderLink`.**]**
 
 ### detach(callback?: (err?: Error) => void): void
 
-**SRS_NODE_AMQP_SENDER_LINK_16_009: [** The `detach` method shall detach the link created by the `amqp10.AmqpClient` underlying object. **]**
+**SRS_NODE_AMQP_SENDER_LINK_06_005: [** If the `SenderLink` is already in the detached state an invocation of `detach` shall immediately invoke the callback with the (potentially) supplied error parameter. **]**
+
+**SRS_NODE_AMQP_SENDER_LINK_16_009: [** The `detach` method shall detach the link created by `rhea`. **]**
+
+**SRS_NODE_AMQP_SENDER_LINK_06_009: [** If `forceDetach` invoked while detaching, the detach will be completed with the error supplied to the `forceDetach` or an error indicating that the `detach` was preempted by the `forceDetach`. **]**
+
+**SRS_NODE_AMQP_SENDER_LINK_06_010: [** An error occurring during a detach will be indicated in the error result of the `detach`. **]**
+
+**SRS_NODE_AMQP_SENDER_LINK_06_011: [** If `detach` invoked while already detaching, it's callback will be invoked with an error.  Whatever caused the original detaching will proceed. **]**
```

Comment:
> Whatever caused the original detaching will proceed. [](start = 130, length = 53)

what does that mean? I'm not sure it adds to the test description or understanding of the situation. try to think of requirements as descriptions for the target as a black box, regardless of the calling context.